src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/Filtering/CompositeSymbolFilter.cs

```diff
@@ -14,7 +14,114 @@ public sealed class CompositeSymbolFilter(params IEnumerable<ISymbolFilter> filt
         /// <summary>
         /// List on inner filters.
         /// </summary>
-        public List<ISymbolFilter> Filters { get; } = new(filters);
+        public List<ISymbolFilter> Filters { get; } = [.. filters];
+
+        /// <summary>
+        /// Creates a composite filter to exclude APIs using the DocIDs provided in the specifed file paths.
+        /// </summary>
+        /// <param name="apiExclusionFilePaths">A collection of paths where the exclusion files should be searched.</param>
+        /// <param name="respectInternals">Whether to include internal symbols or not.</param>
+        /// <param name="includeEffectivelyPrivateSymbols">Whether to include effectively private symbols or not.</param>
+        /// <param name="includeExplicitInterfaceImplementationSymbols">Whether to include explicit interface implementation symbols or not.</param>
+        /// <returns>An instance of the symbol filter.</returns>
+        public static ISymbolFilter GetSymbolFilterFromFiles(string[]? apiExclusionFilePaths,
+                                                                     bool respectInternals = false,
+                                                                     bool includeEffectivelyPrivateSymbols = true,
+                                                                     bool includeExplicitInterfaceImplementationSymbols = true)
+        {
+            DocIdSymbolFilter? docIdSymbolFilter =
```

Comment:
The CompositeSymbolFilter is _one_ implementation of the ISymbolFilter interface and must not know about others, i.e. the `DocIdSymbolFilter`. Even if this these methods are static, they don't belong here. That would violate the separation of concerns.

Reply:
I'll bring back the factory class. If there are methods that belong to a specific class, I'll put them there instead.